### ISSUE TITLE
VS 2015 Update 2 Crash

### ISSUE THREAD

**Issue by fitdev** [jzoss/Git-Source-Control-Provider#54]
I use VS 2015 Update 2 Community. When I enable GIT source control provider from the options, and open a GIT-controlled project VS crashes with this exception:

> Construction of frame content failed. Frame identifier: ST:0:0:{75edecf4-68d8-4b7b-92a9-5915461da6d9} Frame caption: Pending Changes Exception details: System.MissingMethodException: Method not found: 'System.IObservable`1<System.Reactive.EventPattern`1<System.Object>> System.Reactive.Linq.Observable.FromEventPattern(System.Action`1<System.EventHandler>, System.Action`1<System.EventHandler>)'. at GitScc.GitRepository..ctor(String directory) at GitScc.RepositoryManager.GetTrackerForPath(String filename, Boolean setActiveTracker, Boolean createTracker) at GitScc.SccProviderService.get_CurrentTracker() at GitScc.PendingChangesToolWindow.OnToolWindowCreated() at Microsoft.VisualStudio.Shell.Package.CreateToolWindow(Type toolWindowType, Int32 id, UInt32 flags) at Microsoft.VisualStudio.Shell.Package.CreateToolWindow(Type toolWindowType, Int32 id, ProvideToolWindowAttribute tool) at Microsoft.VisualStudio.Shell.Package.FindToolWindow(Type toolWindowType, Int32 id, Boolean create, ProvideToolWindowAttribute tool) at Microsoft.VisualStudio.Shell.Package.CreateToolWindow(Guid& toolWindowType, Int32 id) at Microsoft.VisualStudio.Shell.Package.Microsoft.VisualStudio.Shell.Interop.IVsToolWindowFactory.CreateToolWindow(Guid& toolWindowType, UInt32 id) at Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.ConstructContent()

I have latest GIT for Windows and TotroiseGIT installed, no GIT Extensions.

**Comment by jzoss** [comment]
Interesting .. Thanks for the detailed error.. I'll look into it..

**Comment by jzoss** [comment]
@fitdev I have not been able to recreate your issue yet.. BUT I may have an idea how to fix it.. I attached a special build for you.. Please uninstall what you have and install this.. Might need to turn off autoupdates too.. anyway install it and tell me if it fixes you issue. 

[GitSccProvider.zip](https://github.com/jzoss/Git-Source-Control-Provider/files/326309/GitSccProvider.zip)

**Comment by fitdev** [comment]
Wow! Thanks for such a quick reply and a blazingly fast fix! Seems to have done the trick! I have not yet done extensive testing, but it has not crashed yet where previously it did crash.

As an aside, I am still not certain what difference there is between your GIT Provider and the one built-in to VS 2015 functionality-wise. It is not explained anywhere. (The reason I wanted to try and now am trying yours is that the VS behaves very strangely with directory junctions in the project, effectively not supporting GIT on them, which is a shame, so I am hoping yours will work better with directory junctions. You can see my SO question on this if you are interested: http://stackoverflow.com/questions/37930363/how-to-make-git-treat-directory-junctions-as-regular-folders-on-windows ) Sorry for this aside.

**Comment by jzoss** [comment]
@fitdev I have no idea if that will work.. I've never tired.. :) But Please try it and tell me if it does!

So differences .. It is a fork of a git provider from 2012 and 2013 before built is support.. But.. I liked it better so I forked it and updated it.. It have a few advantages IMHO.. The ui is different , I like using TotroiseGIT UI better that the TFS . Also it works with multiple repos in one solution and has a nice pending changes window. It can update your git ignore ... stuff like that.. 

I'm working on getting gitflow to work as well.

**Comment by fitdev** [comment]
Thanks for such quick responses! Really appreciate it. Yes, it works and did not crash! So whatever you did, it apparently fixed it!

Unfortunately the feature I was hoping for (directory junction support) is not working. I guess VS screws it up somehow. Perhaps it would be a useful feature for your extension to support...

Basic idea is simple: inside your solution which is under GIT control, there maybe folders which are really junctions to other directories (located outside the solution). These folders contain files shared by many different solutions, hence are implemented as junctions. Now, as far as Solution Explorer in VS goes, VS seems to treat these dir. junctions as regular folders as if they resided physically inside solution's folder. However when it comes to GIT it behaves really strangely and effectively does not work.

For example, I tried adding files from such dir. junctions to my repo using regular GIT (no VS involved). I used `git add .` followed by `git commit m "added files"`. And it worked. However as soon as the commit was made the Pending Changes window of your extension showed that file I just added with "Deleted" status and it's icon in the Solution Explorer was shown without any locks/checkmarks (as if it was no longer under source control). This does not make any sense to me. I mean I add a file, and commit in GIT. GIT tells me it did that and all is good, but VS and GIT Provider extension tells me exactly the opposite that file is being deleted from the repo (I guess I still need to make yet another commit in VS now to actually remove teh file from the repo).

I don;t know if it makes any sense at all to you, but perhaps your extension can provide a workaround for this scenario, so that files residing inside directory junctions in VS solutions can be simply manged, checked in and out, commited as files residing in regular folders.
